### ISSUE TITLE
ocr error at specific image with specific model

### ISSUE THREAD

**Issue by safecat** [breezedeus/CnOCR#206]
Image example:
![06_notice_landscape_watermarked](https://user-images.githubusercontent.com/3821031/188781710-26339dee-2786-46e8-b124-5cfac3379c62.jpeg)

use demo page at hugging face:

(default)det:ch_PP-OCRv3_det,onnx rec:densenet_lite_136-fc,onnx result:ok

det:db_mobilenet_v3_small,pytorch rec:densenet_lite_136-fc,onnx result:[ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Non-zero status code returned while running Conv node. Name:'199_nchwc' Status Message: Invalid input shape: {4,0}

det:db_shufflenet_v2_small,pytorch rec:densenet_lite_136-fc,onnx result:[ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Non-zero status code returned while running Conv node. Name:'199_nchwc' Status Message: Invalid input shape: {4,0}

det:db_shufflenet_v2_small,pytorch rec:densenet_lite_136-gru,pytorch result:Given input size: (72x8x1). Calculated output size: (72x4x0). Output size is too small

...

It looks like if I don't use paddlepaddle detect model, there will be some thing wrong with convolutional layer output.

Thanks for all great works!

**Comment by breezedeus** [comment]
yes, it didn't work for some special types of images, such as images with 4 channels. To work around the problem, you can resave your image to more common formats. When I download your image and put it into the demo, it works normally.
![image](https://user-images.githubusercontent.com/6712673/188792056-4536512e-593f-4242-bfbe-d18dd23969a9.png)

**Comment by safecat** [comment]
Thanks for explanation! Please close the issue.

**Comment by safecat** [comment]
Sorry, but it turns out my image is a standard RGB jpeg file, it does not contain 4th channel, and I tried to reproduce the error with file downloaded from this issue page, the error still exists.
I uploaded a screenshot to youtube and here is the link https://www.youtube.com/watch?v=LBgjDB3iaS4 . The error occurred at 15s and numpy/fileinfo showed it's a three channel image at 45s/59s.
The error disappears when I rotate the image to portrait(with exact same format), so I assume the error is image size related.

**Comment by breezedeus** [comment]
thanks for the details. The reason for the error is that the whole image is rotated by 90 degrees. Some detection models do not rotate the detected patches automatically. `det_more_configs={'auto_rotate_whole_image': True}` need to be used when initializing class CnOcr, like this:

<img width="473" alt="image" src="https://user-images.githubusercontent.com/6712673/190683547-c6a8eb31-f6bf-40e9-a478-2ea363bdea86.png">
